src/test/lib/logical_query_plan/lqp_utils_test.cpp

```diff
@@ -212,4 +212,37 @@ TEST_F(LQPUtilsTest, LQPFindModifiedTables) {
   EXPECT_NE(delete_tables.find("node_a"), delete_tables.end());
 }
 
+TEST_F(LQPUtilsTest, CollectSubqueryExpressionsByLQPNestedSubqueries) {
```

Comment:
Could you please add a test in which the same subquery is used twice, making sure that it is only added to the list once? This would be redundant if we could change the list to a map.

Reply:
We use an `unordered_map` from now on. Redundant.